### ISSUE TITLE
PropertyGrid and its elements accessible objects are leaking in memory

### ISSUE THREAD

**Issue by Philip-Wang01** [dotnet/winforms#10495]
### .NET version

.NET 9.0.100-alpha.1.23617.1 + private dlls(build from main branch of WinForms repo)

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Repro on the latest .NET 8.0 (8.0.200-preview.23615.3), previously fixed on .NET 8.0(.NET 8.0.100-preview.2.23101.9).

 Original issue: https://github.com/dotnet/winforms/issues/6943

### Issue description

PropertyGrid accessible objects are left in memory after the PropertyGrid control instance was inspected with accessibility tool (like Accessibility Insights or Inspect) and the form with PropertyGrid is closed, garbage is collected, but accessibility tool is still opened.

Report from WinDbg:

![image](https://github.com/dotnet/winforms/assets/86937911/d41a32fa-be17-4cf9-aca9-7413946ba505)

### Steps to reproduce

Tested sample application:
[PropertyGrid.Sample.zip](https://github.com/dotnet/winforms/files/13700342/PropertyGrid.Sample.zip)

1. Create a parent form with two buttons: one will be opening a child form with the PropertyGrid control and the other will collect the garbage.
2. Start project without debugging(Ctrl+F5).
3. Attaching project to WinDbg and adding symbols folder.
4. Start the Accessibility Insights tool.
5. Press the second button to open the child form with PropertyGrid.
6. Automatic checks for the PropertyGrid control using Accessibility Insight.
7. Close the child form.
8. Clear memory with the first button of the parent form.
9. Don't close the Accessibility Insights tool.
10. Look for PropertyGrid accessible objects left in memory (using WinDbg).

**Comment by lonitra** [comment]
@Philip-Wang01 I have been unable to repro this using 9.0.100-alpha.1.23618.3 in sample project and in the ScratchProject on latest main branch. If you replicate the sample in ScratchProject are you able to repro?

**Comment by Philip-Wang01** [comment]
@lonitra I can reproduce this issue in ScratchProject, please see:

https://github.com/dotnet/winforms/assets/86937911/6e24625b-4bfa-49d1-afb1-2bdf8804a775

**Comment by lonitra** [comment]
Appreciate the video! I see what I did wrong in repro steps.

**Comment by lonitra** [comment]
https://github.com/dotnet/winforms/pull/10479 appears to fix this as I can no longer repro this with its changes. Will need to do more investigation on how we can service this with minimal risk since the difference between .NET 8 and .NET 9 codebase here is nontrivial.

**Comment by Philip-Wang01** [comment]
Verified the issue on 9.0.100-alpha.1.23622.3 with dlls built from main branch of Winforms repo, there is a memory leak when expanding the drop-down list for any property in PropertyGrid control.
![image](https://github.com/dotnet/winforms/assets/86937911/ab4f0219-9a28-4203-81b1-fbd97d9f6593)

**Comment by elachlan** [comment]
@Philip-Wang01 do you see a similar issue with a `ListBox`?

**Comment by Philip-Wang01** [comment]
@elachlan ListBox control does not have this issue in .NET 9.0.

**Comment by elachlan** [comment]
@Philip-Wang01 thanks! I guess we will need to do a bit of debugging and find what the rooted objects are.

**Comment by lonitra** [comment]
Cannot be repro this with recent build `9.0.100-rc.1.24381.9`. Closing for now.

**Comment by MandiMan** [comment]
Verified the issue with .NET 9.0.100-rc.1.24422.10 test pass build that the issue cannot repro:
![image](https://github.com/user-attachments/assets/18e95e00-def5-49f1-aca5-d6a172a4b074)

**Comment by MandiMan** [comment]
Verified this issue with .NET 9.0.100-rc.1.24452.12 test pass build, it has been fixed and the test result is same as above.
